gateway.go

```diff
@@ -162,6 +162,9 @@ func (gw *Gateway) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 	switch state.QType() {
 	case dns.TypeA:
 		m.Answer = gw.A(state, addrs, ttl)
+	case dns.TypeAAAA:
+		m.Ns = []dns.RR{gw.soa(state)}
+		m.Rcode = dns.RcodeNameError
```

Comment:
Line [`return dns.RcodeSuccess, nil`](https://github.com/AbsaOSS/coredns-crd-plugin/blob/main/gateway.go#L180) looks a little bit confusing, as we're resolving with NXDomain error here, but then return `dns.RcodeSuccess, nil`. 
Maybe then we should return `0,nil`, like https://github.com/AbsaOSS/coredns-crd-plugin/blob/main/gateway.go#L159 does?